### DISCUSSION TITLE
In Zephyr, can I develop and debug within Visual Studio?

### DISCUSSION THREAD

**Discussion by MichaelDMcDonnell** [zephyrproject-rtos/zephyr#45984] [General]
In Zephyr, can I develop and debug within Visual Studio? (Not "Visual Code", but rather "Visual Studio").

In other words, I don't want to develop and debug code on an embedded board, I just want to use the PC, Windows, and Visual Studio for development.

**Comment by enthunilu** [comment]
Hi, I was also looking for the same option. However could not find it.
Later I come to know, zephyr supports below IDE's only,
1. VS Code( Using platform IO for all devices or nrf connect for nordic mcu's )
2. Eclipse with zephyr plugin

What I could guess out of this is, when we create a project we are not creating different copy of zephyr source code.
We have a base copy and same is used for all the projects. Basically we are using library only.
This is just a guess and I maybe wrong here.

With this assumption, we can give it a try by importing zephyr code base as well as a part of project and we can try.
Though its a time consuming activity and not sure whether it will give a success.

And with emulators for example QEMU you can debug the developed code.

**Comment by MichaelDMcDonnell** [comment]
Thank you much for your response.

I was looking to just run natively on the PC in Visual Studio (not Visual Code).

It seems that Zephyr is only concerned with building code that will run on a target processor for a particular board that Zephyr supports.

So even when we run with QEMU, we are not running a Windows EXE file, but rather code for a particular target that we emulate on the PC using QEMU.

Do you have any other thoughts?

Thanks!

From: ItachiU ***@***.***>
Sent: Thursday, May 26, 2022 4:18 AM
To: zephyrproject-rtos/zephyr ***@***.***>
Cc: Michael McDonnell ***@***.***>; Author ***@***.***>
Subject: [EXT] Re: [zephyrproject-rtos/zephyr] In Zephyr, can I develop and debug within Visual Studio? (Discussion #45984)

External Email
________________________________

Hi, I was also looking for the same option. However could not find it.
Later I come to know, zephyr supports below IDE's only,

  1.  VS Code( Using platform IO for all devices or nrf connect for nordic mcu's )
  2.  Eclipse with zephyr plugin

What I could guess out of this is, when we create a project we are not creating different copy of zephyr source code.
We have a base copy and same is used for all the projects. Basically we are using library only.
This is just a guess and I maybe wrong here.

With this assumption, we can give it a try by importing zephyr code base as well as a part of project and we can try.
Though its a time consuming activity and not sure whether it will give a success.

And with emulators for example QEMU you can debug the developed code.
